Add `Tablet/Trackpad` 3D navigation preset

RobProductions opened the pull request:
Following up on the discussion from #85331 , if a user enables "emulate 3 button mouse" with the `Godot` preset, they will see the camera always orbit because the `Godot` preset doesn't use any hotkey for orbit. Bringing in "Alt" as a fallback key for 3 button mouse users (which is what the code used to do) would eliminate their ability to customize the orbit button, so instead the idea of a `Tablet/Trackpad` preset seems to work much better.

With this change, 3 button mouse is now tied to the navigation preset, so if you set the preset to `Tablet/Trackpad` it will enable 3 button mouse and also set the orbit key to "Alt" so that Tablet users have better default settings. I also added options and functionality for mouse buttons 4 and 5; none of the presets use this but this should allow users to customize 3D nav further for specific tablet pens or mice.

![image](https://github.com/user-attachments/assets/f5d4904b-7d4f-4a52-bebc-fe7431aee8f3)

With this, tablet and trackpad users can simply change the preset setting rather than changing "Emulate 3 button mouse" which should provide a clearer experience now that everything is customizable. Similarly, if someone changes "Emulate 3 button mouse" from the default the preset will change to `Custom` which should help clarify that they are going outside of the default options. 

~~One question I had about the presets: is it okay to keep pushing down the `Custom` preset to a different number if we add more presets that go above it in the UI? I numbered them in this commit for clarity but say for example we add a `Blender` preset in the future, is it okay for that to become element 4 with `Custom` getting pushed to element 5? Or should I just reorder `Custom` to be at the top now for better futureproofing?~~

Thanks and let me know if there are any questions :)

RobProductions committed:
Add `Tablet/Trackpad` nav preset

This commit adds a new navigation preset called `Tablet/Trackpad` which enables "Emulate 3 Button Mouse" to more quickly set up good default keys for tablet users.

It also adds support for mouse buttons 4 and 5 in the navigation settings which will be helpful if users want to customize 3D navigation further for specific pens/mice.

AThousandShips commented on editor/editor_settings.cpp:
```diff
@@ -811,10 +811,10 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/orbit_mouse_button", 1, "Left Mouse,Middle Mouse,Right Mouse")
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/pan_mouse_button", 1, "Left Mouse,Middle Mouse,Right Mouse")
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/zoom_mouse_button", 1, "Left Mouse,Middle Mouse,Right Mouse")
+	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/navigation_scheme", 0, "Godot,Maya,Modo,Tablet/Trackpad,Custom")
```
This breaks compatibility for anyone who has selected `Custom`

RobProductions replied:
Right, that's why I asked the question about futureproofing in the PR; is there a way to order the enum a specific way in editor settings without changing the int value of `Custom`?

AThousandShips replied:
I'm not sure, I'd say to put it after `Custom` in the values, but order in the same way like so:
```suggestion
	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/navigation_scheme", 0, "Godot:0,Maya:1,Modo:2,Tablet/Trackpad:4,Custom:3")
```

AThousandShips commented:
The new mode needs to be documented in the settings docs

RobProductions commented:
Thanks for the help! I've updated the docs and changed `Custom` back to value 3 so that shouldn't be a compat issue anymore, let me know if that looks good

Mickeon commented on doc/classes/EditorSettings.xml:
```diff
@@ -357,11 +357,12 @@
 			- [b]Godot:[/b] [kbd]Middle mouse button[/kbd] to orbit. [kbd]Shift + Middle mouse button[/kbd] to pan. [kbd]Ctrl + Shift + Middle mouse button[/kbd] to zoom.
 			- [b]Maya:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Middle mouse button[/kbd] to pan, [kbd]Shift + Middle mouse button[/kbd] to pan 10 times faster. [kbd]Alt + Right mouse button[/kbd] to zoom.
 			- [b]Modo:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Alt + Shift + Left mouse button[/kbd] to pan. [kbd]Ctrl + Alt + Left mouse button[/kbd] to zoom.
+			- [b]Tablet/Trackpad:[/b] [kbd]Alt[/kbd] to orbit. [kbd]Shift[/kbd] to pan. [kbd]Ctrl[/kbd] to zoom. The [member editors/3d/navigation/emulate_3_button_mouse] setting will be set to true.
```
```suggestion
			- [b]Tablet/Trackpad:[/b] [kbd]Alt[/kbd] to orbit. [kbd]Shift[/kbd] to pan. [kbd]Ctrl[/kbd] to zoom, enables 3-button mouse emulation mode.
```

Mickeon commented on doc/classes/EditorSettings.xml:
```diff
@@ -357,11 +357,12 @@
 		</member>
 		<member name="editors/3d/navigation/navigation_scheme" type="int" setter="" getter="">
-			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button and modifier controls used to navigate the 3D editor viewport.
+			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button, [member editors/3d/navigation/emulate_3_button_mouse] setting, and modifier keys used to navigate the 3D editor viewport.
```
```suggestion
			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button and modifier keys used to navigate the 3D editor viewport.
```

Mickeon commented on doc/classes/EditorSettings.xml:
```diff
@@ -357,11 +357,12 @@
 			- [b]Maya:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Middle mouse button[/kbd] to pan, [kbd]Shift + Middle mouse button[/kbd] to pan 10 times faster. [kbd]Alt + Right mouse button[/kbd] to zoom.
 			- [b]Modo:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Alt + Shift + Left mouse button[/kbd] to pan. [kbd]Ctrl + Alt + Left mouse button[/kbd] to zoom.
+			- [b]Tablet/Trackpad:[/b] [kbd]Alt[/kbd] to orbit. [kbd]Shift[/kbd] to pan. [kbd]Ctrl[/kbd] to zoom. The [member editors/3d/navigation/emulate_3_button_mouse] setting will be set to true.
 			See also [member editors/3d/navigation/orbit_mouse_button], [member editors/3d/navigation/pan_mouse_button], [member editors/3d/navigation/zoom_mouse_button], and [member editors/3d/freelook/freelook_navigation_scheme].
```
```suggestion
			See also [member editors/3d/navigation/orbit_mouse_button], [member editors/3d/navigation/pan_mouse_button], [member editors/3d/navigation/zoom_mouse_button], and [member editors/3d/freelook/freelook_navigation_scheme], and [member editors/3d/navigation/emulate_3_button_mouse].
```

Mickeon reviewed (commented):
I don't think it's worth being so explicit over this one editor setting. The other affected ones are listed at the bottom of the description.

Calinou reviewed (approved):
Tested locally on a MacBook Pro, it works as expected. The shortcuts with this preset make sense to me.

Repiteo commented:
Thanks!
